backend/icpy/services/preview_service.py

```
@@ -0,0 +1,894 @@
+            await self.message_broker.publish("preview.created", {
+                "preview_id": preview_id,
+                "project_type": detected_type.value,
+                "files_count": len(files),
+                "content_hash": content_hash,
+                "timestamp": time.time()
+            })
+            
+            return preview_id
+            
+        except Exception as e:
+            logger.error(f"Failed to create preview {preview_id}: {e}")
```

Ruff (0.12.2)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)